app/controllers/comments_controller.rb

```diff
@@ -3,26 +3,27 @@ class CommentsController < ApplicationController
   http_basic_authenticate_with name: "dhh", password: "secret"
 
   def create
-    @article = Article.find(params[:article_id])
-    @comment = @article.comments.build(comment_params)
-
-    if @comment.save
-      redirect_to @article
+    @comment = Comments::Create.call(comment_params)
+    @article = @comment.article
+    if @comment.errors.any?
+      render 'articles/show'
     else
-      render @article, status: :unprocessable_entity
+      redirect_to @comment.article, status: :see_other
     end
   end
 
   def destroy
-    @article = Article.find(params[:article_id])
-    @comment = @article.comments.find(params[:id])
-    @comment.destroy
-    redirect_to article_path(@article), status: :see_other
+    @destroy_errors = Comments::Destroy.call(params[:article_id], params[:id])
```

Comment:
In terms of naming it looks weird: looks like you waiting only errors in response here. 
better looking will be:
```
@destroyed_comment = Comments::Destroy.call(params[:article_id], params[:id])
if @destroyed_comment.errors.any?
...
```